core/imagedownloader.cpp

```diff
@@ -85,26 +85,31 @@ static bool hasVideoFileExtension(const QString &filename)
 #ifdef LIBRAW_SUPPORT
 QImage fetchRawThumbnail(const QString &filename)
 {
-	LibRaw raw; // Might think about reusing that, one instance per thread
-
-	// TODO: Convert filename to UTF-16 for windows
-	if (raw.open_file(qPrintable(filename)) != LIBRAW_SUCCESS ||
-	    raw.unpack_thumb() != LIBRAW_SUCCESS) {
+	// Might think about reusing that, one instance per thread
+	// This is quite heavy, so better not allocate it on the stack
+	LibRaw *raw =  new LibRaw;
+
+	       // TODO: Convert filename to UTF-16 for windows
+	if (raw->open_file(qPrintable(filename)) != LIBRAW_SUCCESS ||
+	    raw->unpack_thumb() != LIBRAW_SUCCESS) {
+		delete raw;
 		return QImage();
 	}
 
-	switch (raw.imgdata.thumbnail.tformat) {
+	switch (raw->imgdata.thumbnail.tformat) {
 	case LIBRAW_THUMBNAIL_JPEG: {
 		QImage res;
-		res.loadFromData(reinterpret_cast<unsigned char *>(raw.imgdata.thumbnail.thumb),
-				 raw.imgdata.thumbnail.tlength);
+		res.loadFromData(reinterpret_cast<unsigned char *>(raw->imgdata.thumbnail.thumb),
+				 raw->imgdata.thumbnail.tlength);
+		delete raw;
 		return res;
 	}
 	case LIBRAW_THUMBNAIL_BITMAP:
-		return QImage(reinterpret_cast<unsigned char *>(raw.imgdata.thumbnail.thumb),
-			      raw.imgdata.thumbnail.twidth, raw.imgdata.thumbnail.theight,
+		return QImage(reinterpret_cast<unsigned char *>(raw->imgdata.thumbnail.thumb),
```

Comment:
With your original code, this would not free the `raw` object. Is this on purpose? This surprising behavior would require a fat comment.